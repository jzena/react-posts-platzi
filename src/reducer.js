const initialState = {
    posts: {
        page: 1,
        entities: [],
    },
    comments: [],
    user: {},
};

const action = {
    type: 'SET_POST',
    payload: {},//data que necesitamos para hacer la acción
    //meta: {},//cualquier información extra que necesitamos
    //error: true,//default en redux
}

function reducer(state = initialState, action = {}) {
    switch (action.type) {
        case 'SET_POST':
            return Object.assign({}, state, {
                posts: Object.assign({}, state.posts, {
                    entities: state.posts.entities.concat(action.payload),
                }),
            });
        default:
            return state;
    }
}