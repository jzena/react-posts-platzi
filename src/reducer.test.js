import reducer from './reducer';
import actions from './actions';

test('reducer - SET_COMMENTS', () => {
    //throw new Error('test');
    expect(
        reducer(
            undefined,
            actions.setComments([{ id: 1 }, { id: 3 }])
        )
    ).toMatchSnapshot();
})