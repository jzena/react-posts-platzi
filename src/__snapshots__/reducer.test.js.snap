// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer - SET_COMMENTS 1`] = `
Immutable.Map {
  posts: Immutable.Map {
    page: 1,
    entities: Immutable.Map {
    },
  },
  comments: Immutable.Map {
    1: Immutable.Map {
      id: 1,
    },
    3: Immutable.Map {
      id: 3,
    },
  },
  users: Immutable.Map {
  },
}
`;
